File: smile-clinic-back/smile-clinic/src/main/java/com/smile/clinic/smile_clinic/infrastructure/adapters/input/rest/models/AppointmentDTO.java
package com.smile.clinic.smile_clinic.infrastructure.adapters.input.rest.models;

import com.smile.clinic.smile_clinic.domain.models.appointments.AppointmentState;
import com.smile.clinic.smile_clinic.domain.models.patients.Patient;
import com.smile.clinic.smile_clinic.domain.models.users.DentistData;
import com.smile.clinic.smile_clinic.domain.models.users.User;
import com.smile.clinic.smile_clinic.infrastructure.adapters.input.rest.models.patientsDTO.PatientDTO;
import com.smile.clinic.smile_clinic.infrastructure.adapters.input.rest.models.usersDTO.RegisteredUserDTO;
import lombok.*;

import java.time.LocalDateTime;

@Builder
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class AppointmentDTO {
    private Long id;
    private double duration;
    private String visitPurpose;
    private LocalDateTime dateTime;
    private DentistData user;
    private PatientDTO patient;
}

File: smile-clinic-front/smile-clinic/src/app/models/AppointmentDTO.ts
import { DentistDataDTO } from "./DentistDataDTO";
import { PatientDTO } from "./PatientDTO";
import { RegisteredUserDTO } from "./RegisteredUserDTO";
import { userData } from "./userData";

export interface AppointmentDTO {
  id: string;
  duration: number;
  visitPurpose: string;
  appointmentState: string;
  dateTime: string; //DateTime
  user: DentistDataDTO;
  patient: PatientDTO;
}